src/script_opt/Stmt.cc

```diff
@@ -919,6 +919,12 @@ StmtPtr InitStmt::DoReduce(Reducer* c)
 	return ThisPtr();
 	}
 
+StmtPtr AssertStmt::Duplicate()
+	{
+	// Is this right?
+	return SetSucc(new AssertStmt(cond->Duplicate(), msg ? msg->Duplicate() : nullptr));
```

Comment:
@vpax - mind skimming the `Duplicate()` implementation over?

Reply:
That looks fine - thanks!